core/src/main/java/io/lionweb/lioncore/java/model/impl/DynamicNode.java

```diff
@@ -70,14 +71,40 @@ public boolean equals(Object o) {
     }
     DynamicNode that = (DynamicNode) o;
     return Objects.equals(id, that.id)
-        && Objects.equals(parent, that.parent)
```

Comment:
Are these changes checked by unit tests, potentially implicitly through `assertEquals`?

Reply:
I added a test covering the issue you pointed out above. In DynamicNodeTest we had other tests for equality, so we should have a decent coverage for this method